astropy/io/ascii/core.py

```diff
@@ -665,10 +665,8 @@ def write_comments(self, lines, meta):
 
     def write(self, lines: list[str]) -> None:
         if self.start_line is not None:
-            for i, spacer_line in zip(
-                range(self.start_line), itertools.cycle(self.write_spacer_lines)
-            ):
-                lines.append(spacer_line)
+            for spacer_line in itertools.islice(itertools.cycle(self.write_spacer_lines), self.start_line):
+                lines.append(spacer_line.copy())
```

Comment:
Please avoid any code logic change in this ruff PR. Your changes should mostly be limited to either changing a loop variable to `_`, or in some cases eliminating an unnecessary `enumerate`.